api/api.go

```diff
@@ -206,10 +206,7 @@ func handleRefRoute(m *mux.Router, p refRouteParams, f http.HandlerFunc) {
 func NewServerRoutes(s Server) *mux.Router {
 	cfg := s.GetConfig()
 
-	m := s.Mux
-	if m == nil {
-		m = mux.NewRouter()
-	}
+	m := s.Instance.GiveAPIServer(s.Middleware, []string{"dataset.get"})
 	m.Use(corsMiddleware(cfg.API.AllowedOrigins))
```

Comment:
This line will take the lib supplied mux and allow CORS given OPTIONS requests pass to it.